command/agent.go

```diff
@@ -261,6 +261,17 @@ func (c *AgentCommand) Run(args []string) int {
 		return 1
 	}
 
+	serverHealth, err := client.Sys().Health()
+	if err == nil {
+		// We don't exit on error here, as this is not worth stopping Agent over
+		serverVersion := serverHealth.Version
+		agentVersion := version.GetVersion().VersionNumber()
+		if serverVersion != agentVersion {
+			c.UI.Info("==> Note: Vault Agent is not running with the same version as Vault itself, we recommend you upgrade! " +
+				fmt.Sprintf("Vault Agent version: %s, Vault version: %s", agentVersion, serverVersion))
```

Comment:
Just a thought, but could we just note that the versions do not match and leave it at that? 

e.g.

```suggestion
			c.UI.Info("==> Note: Vault Agent version does not match Vault server. " +
				fmt.Sprintf("Agent version: %s, Vault version: %s", agentVersion, serverVersion))
```

This was my thinking:

* What if we're running a cluster with different versions of Vault, we can only report on the initial node we connect to (bad run different versions in a cluster, I know)?
* What if the Vault server version is older than Agent?
* What if they're different by a point release?

Should we be recommending an upgrade if `!=`? I'm just not sure, so that's why I figured, just send the `info` to the log and let them figure out what to do. :)

Reply:
The reason I worded it how I did was kinda to imply that we recommend upgrading Agent if it's lower or Server if that's lower, but I do like this suggestion - it keeps it more agnostic but would still hopefully make customers think "hmm" if Agent is e.g. 1.0 and Server is 1.10

Reply:
Updated!